### PR TITLE
chore: add verifiable build, reenable programs to be registered signer check

### PR DESCRIPTION
**Pull request by ananas-block**
Changes:
- add verifiable build
- reenable program signer check for programs to be registered
- remove commented check for protocol config init (it is already inited on mainnet and annoying in tests)

### CODE REVIEW

File: programs/account-compression/src/instructions/register_program.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ pub struct RegisterProgramToGroup<'info> {
     #[account( mut, constraint= authority.key() == group_authority_pda.authority @AccountCompressionErrorCode::InvalidAuthority)]
     pub authority: Signer<'info>,
     /// CHECK: TODO: check that upgrade authority is signer.
```
Comment (SwenSchaeferjohann) [inline programs/account-compression/src/instructions/register_program.rs:19]: remove todo ?

File: programs/registry/src/account_compression_cpi/register_program.rs [hunk 2]
```diff
@@ -25,7 +25,7 @@ pub struct RegisterProgram<'info> {
     pub registered_program_pda: AccountInfo<'info>,
     /// CHECK: (account compression program). TODO: check that a signer is the upgrade authority.
```
Comment (SwenSchaeferjohann) [inline programs/registry/src/account_compression_cpi/register_program.rs:26]: remove todo?

Comment (ananas-block) [reply programs/registry/src/account_compression_cpi/register_program.rs:26]: want to keep this todo for a next release

File: programs/account-compression/src/instructions/register_program.rs (hunk 1 above)
Comment (ananas-block) [reply programs/account-compression/src/instructions/register_program.rs:19]: want to keep this todo for a next release

**Review by SwenSchaeferjohann — APPROVED** [review]

**Pull request merged**